app/models/concerns/auto_favourite.rb

```diff
@@ -0,0 +1,46 @@
+module AutoFavourite
+  extend ActiveSupport::Concern
+  
+  included do
+    after_create  :mark_recent_usage
+    after_update  :mark_recent_usage
+    after_destroy :remove_from_favourites
+    
+    scope :recently_used, -> (user_id) {
+      joins(%{
+        LEFT JOIN user_favourites ON user_favourites.favourite_type = '#{self.class.name}' AND user_favourites.user_id = #{user_id}
+      }).order('user_favourites.updated_at').limit(20)
```

Comment:
don't think `limit` has its place in a scope. That's the responsibility of whoever/wherever calls the scope